sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -410,13 +412,44 @@ private GatherResults(Coder<ResultT> resultCoder) {
       } else {
         // Pass results via a side input rather than reshuffle, because we need to get an empty
         // iterable to finalize if there are no results.
-        return input
-            .getPipeline()
-            .apply(Reify.viewInGlobalWindow(input.apply(View.asList()), ListCoder.of(resultCoder)));
+        return input.apply("ToList", Combine.globally(new ToListCombineFn<>()));
       }
     }
   }
 
+  public static class ToListCombineFn<ResultT>
```

Comment:
nit: This should have been private class since its an implementation detail and didn't need to expose this to users of the WriteFiles transform.

Reply:
You're right, I just copy pasted the code from Jira. It was my first PR in Beam and I wanted to get the workflow without digging the code proposed.